File: java/indexes.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class indexes {

    public static void main(String[] args) {

		try {
			if(args.length < 1) {
				System.out.println("syntax: indexes.py [list | show indexname | clean indexname | create indexname | delete indexname]");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);

	        if(args[0].equalsIgnoreCase("list")) {
				//LAB: list all indexes by name to STDOUT
			}
			else if(args[0].equalsIgnoreCase("show")) {
				Index index = //LAB: get the index named in args[1]
				//LAB: print all index attributes
			}
			else if(args[0].equalsIgnoreCase("clean")) {
				Index index = //LAB: get the index named in args[1]
				//LAB: clean the index
				System.out.println("  index " + index.getName() + " cleaned.");
			}
			else if(args[0].equalsIgnoreCase("create")) {
				Index index = //LAB: create the index named in args[1]
				System.out.println("  index " + index.getName() + " created.");
			}
			else if(args[0].equalsIgnoreCase("delete")) {
				//LAB: delete the index named in args[1]
				System.out.println("  " + args[1] + " deleted.");
			}
			else {
				throw new Exception("Unknown command \"" + args[0] + "\"");
			}
			
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}

File: python/indexes.py

import splunklib.client as client
import utils
import sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
try: 
    if len(sys.argv) == 1:
        print "syntax: indexes.py [list | show indexname | clean indexname | delete indexname | create indexname]"
        sys.exit()
    service = client.connect(**opts.kwargs)

    if sys.argv[1] == "list":
        #LAB: print all index names to STDOUT
    elif sys.argv[1] == "show":
        index = #LAB: get the index named in sys.argv[2]
        #LAB: print all index attributes to STDOUT
    elif sys.argv[1] == "create":
        index = #LAB: create the index named in sys.argv[2]
        print "  created index %s" % index.name
    elif sys.argv[1] == "clean":
        index = #LAB: get the index named in sys.argv[2]
        #LAB: clean the index 
        print "  cleaned index %s" % index.name
    elif sys.argv[1] == "delete":
        index = #LAB: get the index named in sys.argv[2]
        #LAB: delete the index
        print "  index %s deleted" % sys.argv[2]
    else:
        raise Exception("unknown command \"%s\"" % sys.argv[1])

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))
